src/ai/backend/manager/container_registry/base.py

```diff
@@ -142,27 +131,44 @@ async def commit_rescan_result(self) -> None:
                     image_row.is_local = is_local
                     image_row.resources = values["resources"]
 
-                session.add_all([
-                    ImageRow(
-                        name=image_ref.canonical,
-                        registry=image_ref.registry,
-                        registry_id=await self.get_registry_id(
-                            registry_name=image_ref.registry,
-                            project=image_ref.name.split("/")[0] if "/" in image_ref.name else None,
-                        ),
-                        image=image_ref.name,
-                        tag=image_ref.tag,
-                        architecture=image_ref.architecture,
-                        is_local=is_local,
-                        config_digest=v["config_digest"],
-                        size_bytes=v["size_bytes"],
-                        type=ImageType.COMPUTE,
-                        accelerators=v.get("accels"),
-                        labels=v["labels"],
-                        resources=v["resources"],
-                    )
-                    for image_ref, v in _all_updates.items()
-                ])
+                registry_cache = {}
+
+                for image_ref, v in _all_updates.items():
+                    if image_ref.registry not in registry_cache:
+                        query = sa.select([
+                            ContainerRegistryRow.id,
+                            ContainerRegistryRow.project,
+                        ]).where(ContainerRegistryRow.registry_name == image_ref.registry)
+
+                        registries = (await session.execute(query)).fetchall()
+                        registry_cache[image_ref.registry] = registries
+                    else:
+                        registries = registry_cache[image_ref.registry]
+
+                    for registry in registries:
+                        if image_ref.name.startswith(registry.project):
+                            if not image_ref.name.split(registry.project)[1].startswith("/"):
```

Comment:
The reason for writing this branch logic is to handle cases where, for example, image_ref.name is "namespace/project" and registry.project is "namespace/project2".

Of course, in this case, since the projects are different, this registry_id should not be selected.